src/loginpage/LoginForm.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import styled from 'styled-components';
+import Label from '../components/Label';
+import Input from '../components/Input';
+import SubmitButton from '../components/SubmitButton';
+
+const StyledLoginForm = styled.form``;
+
+export default function LoginForm({
+  userGroups,
+  onLogin,
+  history,
+  onIncorrectLoginData
+}) {
+  function handleLoginFormSubmit(
+    event,
+    userGroups,
+    onLogin,
+    history,
+    onIncorrectLoginData
```

Comment:
here the same as above. `onLogin` `history` `onIncorrectLoginData` `userGroups` are globally available so you dont need to put them here as arguments and parameters in line 49-52. The only thing you need here is `event` because its the only "new" parameter you get out of `onSubmit` and call the function `onSubmit={event => handleLoginFormSubmit(event)}`